model/scope.go

```diff
@@ -0,0 +1,168 @@
+package model
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+type Scope []string
```

Comment:
It would be better to have typed strings as a scope. And then have a slice of scopes.

```
type Scope string
const (
	ScopeChannelJoin   Scope = "channels:join"
	ScopeMessageRead Scope = "message:read"
)
```

And then use either `Scope` for a single scope, or `[]Scope` for multiple scopes. This also gives you extra type safety against random strings getting into scopes.